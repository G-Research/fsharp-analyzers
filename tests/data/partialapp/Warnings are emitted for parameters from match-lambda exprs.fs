module M

let myFunc x y = x + y
let myGenFunc<'t> x y z = x + y + z

let _ =
    myFunc
        (23
            |> function
                | 1 -> 23
                | _ -> 42)
        111 // should not warn

let _ =
    myFunc (    // should warn
        23
        |> function
            | 1 -> 23
            | _ -> 42
    )

let _ =
    (23
        |> function
            | 1 -> myFunc   // should not warn because no app
            | _ -> myGenFunc<int> 23)   // should warn
        42

let _ =
    (23
        |> function
            | 1 -> myFunc 23 42
            | _ -> myGenFunc<int> 23 42 101) // should not warn
