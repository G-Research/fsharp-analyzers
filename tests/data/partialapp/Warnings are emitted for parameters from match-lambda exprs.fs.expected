GRA-PARTAPP-001 | Warning | (15,4 - 15,10) | partial application should not be used: myFunc at (15,4--15,10)
GRA-PARTAPP-001 | Warning | (26,19 - 26,28) | partial application should not be used: myGenFunc at (26,19--26,28)
